packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1711,8 +1712,35 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 return { ...attemptData, error };
             }
 
+            const summary = summarizeResult.summary.tree;
+            const summaryTree = summary[channelsTreeName];
+            let totalDataStoreCount = 0;
+            let unreferencedDataStoreCount = 0;
+            let handleCount = 0;
+            if (summaryTree.type === SummaryType.Tree) {
```

Comment:
I'd rather go for assert() here vs. runtime IF check, as not having tree there likely points to serious issue.
Same for next IF.

Reply:
I think we need to handle the case when this.disableIsolatedChannels is true

Reply:
As mentioned here: https://github.com/microsoft/FluidFramework/pull/6484#discussion_r655580096

Reply:
Sounds good :)

Reply:
That said, having some reference in comments to it would be nice, such that if/when we decide to remove disableIsolatedChannels , we can find this code and adjust.

Reply:
does this still apply to dataStore.ts ?

Reply:
Yes, nothing changes by moving code around :)

Reply:
but in dataStore.ts , context summary can be either tree or handle,